includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -210,21 +210,33 @@ public function get_button_height() {
 	 *
 	 * @param object $product WC_Product_* object.
 	 * @return mixed Total price.
+	 *
+	 * @throws Invalid_Price_Exception Whenever a product has no price.
 	 */
 	public function get_product_price( $product ) {
-		$product_price = $product->get_price();
-
 		// If prices should include tax, using tax inclusive price.
 		if ( ! $this->prices_exclude_tax() ) {
-			$product_price = wc_get_price_including_tax( $product );
+			$base_price = wc_get_price_including_tax( $product );
+		} else {
+			$base_price = $product->get_price();
 		}
 
 		// Add subscription sign-up fees to product price.
-		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) ) {
-			$product_price = $product_price + WC_Subscriptions_Product::get_sign_up_fee( $product );
+		$sign_up_fee        = 0;
+		$subscription_types = [
+			'subscription',
+			'subscription_variation',
+		];
+		if ( in_array( $product->get_type(), $subscription_types, true ) && class_exists( 'WC_Subscriptions_Product' ) ) {
+			// When there is no sign-up fee, `get_sign_up_fee` falls back to an int 0.
+			$sign_up_fee = WC_Subscriptions_Product::get_sign_up_fee( $product );
 		}
 
-		return $product_price;
+		if ( ! is_numeric( $base_price ) || ! is_numeric( $sign_up_fee ) ) {
+			throw new Invalid_Price_Exception( __( 'Express checkout does not support products without prices! ', 'woocommerce-payments' ) );
```

Comment:
Do we want to log additional properties here by mentioning the product details? Would be easier to debug in case this starts showing up in logs.

Reply:
Done here: https://github.com/Automattic/woocommerce-payments/pull/4649/commits/b973eb30a00bd00bd8e99443ba8e7b2deec98c18 (just the product ID, but that should be more than clear enough). I considered adding the product name, but I think variations don't have proper names.